### ISSUE TITLE
expression in sizeof is evaluated if it is a constant expression

### ISSUE THREAD

**Issue by dwightguth** [kframework/c-semantics#113]
```
int main() {
  size_t x = sizeof(*((char *)0));
  return 0;
}
```

This program should succeed. However, we report a false positive of a null pointer read because we are evaluating the constant expression associated with dereferencing a null pointer. Something needs to be done so that we can compute the type of sizeof expressions without requiring that the expression is evaluated for possible undefined behaviors.

**Comment by chathhorn** [comment]
FWIW, the argument usually isn't evaluated (except for variable-length array types), it's just that the undefinedness check is getting triggered while calculating the type.

**Comment by dwightguth** [comment]
Fixed in #115
